op-batcher/batcher/channel_builder.go

```diff
@@ -96,10 +98,20 @@ func (cc *ChannelConfig) Check() error {
 	return nil
 }
 
-// InputThreshold calculates the input data threshold in bytes from the given
-// parameters.
-func (c ChannelConfig) InputThreshold() uint64 {
-	return uint64(float64(c.TargetNumFrames) * float64(c.TargetFrameSize) / c.ApproxComprRatio)
+func (cc *ChannelConfig) NewCompressor() (derive.Compressor, error) {
+	switch cc.CompressorKind {
+	case flags.ShadowCompressorKind:
+		return NewShadowCompressor(
+			cc.TargetFrameSize,
+			cc.TargetNumFrames,
+		)
+	default:
```

Comment:
This code works, but the default statement is doing a lot of work to handle the cases where the compressor kind is not set while constructing the batcher in process.